sdk/storage/Azure.Storage.Common/src/Shared/LazyLoadingReadOnlyStream.cs

```diff
@@ -186,7 +186,7 @@ public async Task<int> ReadInternal(byte[] buffer, int offset, int count, bool a
                 }
             }
 
-            if (_bufferPosition == 0 || _bufferPosition == _bufferLength || _bufferInvalidated)
+            if (_bufferPosition == _bufferLength || _bufferInvalidated)
```

Comment:
Are we sure changing this and the line below fixed the existing bug without introducing another one?